app/models/post.rb

```diff
@@ -1007,6 +1007,25 @@ def link_post_uploads(fragments: nil)
       upload = nil
       upload = Upload.find_by(sha1: sha1) if sha1.present?
       upload ||= Upload.get_from_url(src)
+
+      # Link any video thumbnails
+      if upload.present?
+        # Video thumbnails have the filename of the video file sha1 with a .png or .jpg extention.
```

Comment:
```suggestion
        # Video thumbnails have the filename of the video file sha1 with a .png or .jpg extension.
```

This is interesting. Maybe we can restrict this to video-type uploads only?